app/src/main/java/com/edulectronics/tinycircuit/Controllers/WireController.java

```diff
@@ -45,12 +59,46 @@ public void makeWire(Component component, MotionEvent event) {
                             getConnectionPoint(firstComponent, firstOrientation),
                             getConnectionPoint(component, secondOrientation));
                     connecting = false;
-                    wireView.invalidate();
+
+                    redrawWires();
                 }
             }
         }
     }
 
+    // Redraw all the wire views in the grid
+    public void redrawWires() {
+        // First clear the existing wires
+        clearWires();
+
+        // ... and rebuild.
+        for (Connection c : circuitController.getAllConnections()) {
+            if (c != null) {
+                Point startPoint = coordinateHelper.getNodeLocation(
+                        c.pointA.getParentComponent().position,
+                        c.pointA.orientation);
+
+                Point endPoint = coordinateHelper.getNodeLocation(
+                        c.pointB.getParentComponent().position,
+                        c.pointB.orientation
+                );
+
+                setLines(c, startPoint, endPoint);
+            }
+        }
+        for (Wire wire : this.wires) {
+            wire.invalidate();
+        }
+    }
+
+    private void clearWires() {
+        for (Wire wire : wires) {
+            DrawerLayout parentLayout = (DrawerLayout)((Activity)context).findViewById(R.id.wires);
```

Comment:
Het is volgnes mij onnodig om de parentlayout voor elk draad opnieuw aan te maken